jupyter_client/session.py

```diff
@@ -928,10 +928,14 @@ def deserialize(self, msg_list, content=True, copy=True):
         if not len(msg_list) >= minlen:
             raise TypeError("malformed message, must have at least %i elements"%minlen)
         header = self.unpack(msg_list[1])
-        message['header'] = extract_dates(header)
+
+        message['header'] = header
+        message['header']['date'] = parse_date(message['header']['date'])
         message['msg_id'] = header['msg_id']
         message['msg_type'] = header['msg_type']
-        message['parent_header'] = extract_dates(self.unpack(msg_list[2]))
+        message['parent_header'] = self.unpack(msg_list[2])
+        if 'date' in message['parent_header']:
```

Comment:
This will break current stable ipyparallel, which currently relies on the behavior of parsing any date-like strong, but I can deal with that. I never should have made it find and parse any valid date objects!

I suspect that parsing just the date will break exactly as many things as disabling parsing entirely, though. Which is to say: ipyparallel and probably nothing else.